cli/src/pcluster/templates/imagebuilder_stack.py

```diff
@@ -0,0 +1,172 @@
+        imagebuilder.CfnInfrastructureConfiguration(
+            self,
+            id="PClusterImageInfrastructureConfiguration",
+            name="PCluster-Image-Infrastructure-Configuration",
+            instance_profile_name=core.Fn.ref("InstanceProfile"),
+            terminate_instance_on_failure=build_config.terminate_on_failure,
+            instance_types=instance_types,
+        )
+
+        # Define ami build instance ebs
+        ebs = imagebuilder.CfnImageRecipe.EbsInstanceBlockDeviceSpecificationProperty(
+            delete_on_termination=image_config.root_volume.delete_on_termination,
+            volume_size=image_config.root_volume.size,
+            volume_type="gp2",
+        )
+
+        cloudformation_dir = "../../../../cloudformation/imagebuilder/"
+
+        # Component
+        if dev_settings_config and dev_settings_config.update_and_reboot:
+            imagebuilder.CfnComponent(
+                self,
+                id="UpdateAndRebootComponent",
+                name="UpdateAndReboot",
+                version="0.0.1",
+                description="Update OS and Reboot",
+                change_description="First version",
+                platform="Linux",
+                data=core.Fn.sub(get_component_data(cloudformation_dir, "update_and_reboot.yaml")),
+            )
+            imagebuilder.CfnComponent(
+                self,
+                id="PClusterComponent",
+                name="PCluster",
+                version="0.0.1",
+                description="Bake PCluster AMI",
+                change_description="First version",
+                platform="Linux",
+                data=core.Fn.sub(get_component_data(cloudformation_dir, "pcluster_install.yaml")),
+            )
+
+            components = [
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("UpdateAndReboot")
+                ),
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("PClusterComponent")
+                ),
+            ]
+        else:
+            imagebuilder.CfnComponent(
+                self,
+                id="PClusterComponent",
+                name="PCluster",
+                version="0.0.1",
+                description="Bake PCluster custom AMI",
+                change_description="First version",
+                platform="Linux",
+                data=core.Fn.sub(get_component_data(cloudformation_dir, "pcluster_install.yaml")),
+            )
+
+            components = [
+                imagebuilder.CfnImageRecipe.ComponentConfigurationProperty(
+                    component_arn=core.Fn.ref("PClusterComponent")
+                )
+            ]
+
+        # ImageRecipe
+        imagebuilder.CfnImageRecipe(
+            self,
+            id="PClusterImageRecipe",
+            name="PCluster-2-10-1" + str(hash(imagebuild)),
```

Comment:
if you need the version let's use the utils.get_pclsuter_version and convert the value. We should have a single place in the code with the 2.10.1 string, to be easier bump the version for the new releases.